curator-framework/src/test/java/org/apache/curator/framework/imps/TestReconfiguration.java

```diff
@@ -313,6 +313,38 @@ public void testNewMembers() throws Exception
         }
     }
 
+    @Test
+    public void testConfigToConnectionStringNormal() throws Exception
+    {
+        String config = "server.1=10.1.2.3:2888:3888:participant;10.2.3.4:2181";
+        String configString = EnsembleTracker.configToConnectionString(toQuorumVerifier(config.getBytes()));
+        Assert.assertEquals("10.2.3.4:2181", configString);
+    }
+
+    @Test
+    public void testConfigToConnectionStringNoClientAddr() throws Exception
+    {
+        String config = "server.1=10.1.2.3:2888:3888:participant;2181";
+        String configString = EnsembleTracker.configToConnectionString(toQuorumVerifier(config.getBytes()));
+        Assert.assertEquals("10.1.2.3:2181", configString);
+    }
+
+    @Test
+    public void testConfigToConnectionStringWildcardClientAddr() throws Exception
+    {
+        String config = "server.1=10.1.2.3:2888:3888:participant;0.0.0.0:2181";
```

Comment:
I seem to be getting a NPE with 3.5.3-beta when attempting to construct the TestingZooKeeperServer:

    java.lang.NullPointerException
	    at org.apache.zookeeper.server.quorum.QuorumPeerMain.runFromConfig(QuorumPeerMain.java:158)
	    at org.apache.curator.test.TestingZooKeeperServer$1.run(TestingZooKeeperServer.java:150)
	    at java.lang.Thread.run(Thread.java:745)

It looks like is a known issue, but I don't understand the fix version of 3.4.0 when it occurred on 3.5.3-beta only - I assume that should be 3.5.4-beta:

https://issues.apache.org/jira/browse/CURATOR-409

I'm not really sure how I should proceed here. Updating the pom would break a lot of the tests, and I can't add IPv6 tests without the updated Zookeeper version supporting IPv6.